app/src/main/java/com/github/multimatum_team/multimatum/activity/AddDeadlineActivity.kt

```diff
@@ -261,43 +277,78 @@ class AddDeadlineActivity : AppCompatActivity() {
         if (titleDeadline == "") {
             Toast.makeText(this, getString(R.string.enter_a_title), Toast.LENGTH_SHORT).show()
         } else {
-            // Add the deadline
-            val deadline = Deadline(
-                titleDeadline,
-                DeadlineState.TODO,
-                selectedDate,
-                textDescription.text.toString()
-            )
-
-            val notificationsTimes = retrieveNotificationsTimes()
-
-            Toast.makeText(this, getString(R.string.deadline_created), Toast.LENGTH_SHORT).show()
-
             // Reset the text input for future use
             textTitle.text = ""
-
-            deadlineListViewModel.addDeadline(deadline) {
-                DeadlineNotification.editNotification(it, deadline, notificationsTimes, this)
-                finish()
+            //loading bar
+            progressBar.visibility = View.VISIBLE
+            //start upload
+            uploadPdfToFirebase(pdfData) { ref ->
+                //hide loading bar
+                progressBar.visibility = View.GONE;
+
+                // create the deadline
+                val deadline = Deadline(
+                    titleDeadline,
+                    DeadlineState.TODO,
+                    selectedDate,
+                    textDescription.text.toString(),
+                    pdfPath = ref
+                )
+                //get notification setting
+                val notificationsTimes = retrieveNotificationsTimes()
+
+                //send toast
+                Toast.makeText(this, getString(R.string.deadline_created), Toast.LENGTH_SHORT)
+                    .show()
+
+                //add the deadline and finish the activity
+                deadlineListViewModel.addDeadline(deadline) {
+                    DeadlineNotification.editNotification(it, deadline, notificationsTimes, this)
+                    finish()
+                }
             }
+
         }
     }
+
+    //code to be executed when a pdf has been chosen
     private val startForResult =
         registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result: ActivityResult ->
             if (result.resultCode == Activity.RESULT_OK) {
-                val intent = result?.data!!
-                val pdfUri = intent.data!!
-                val path: String = pdfUri.toString()
+                pdfData = result.data!!.data!!
```

Comment:
Can you explain what is going on here ?

Reply:
Well this is how I have to retrieve the Uri. It's a bit odd to have `.data` twice in a row, but it's not from me.
The first one is to get an `Intent` from `ActivityResult`
And the second one is to get an `Uri` from the `ActivityResult`